policy-controller/k8s/status/src/index.rs

```diff
@@ -0,0 +1,246 @@
+                        namespace: Some(parent_reference_id.namespace.clone()),
+                        name: parent_reference_id.name.clone(),
+                        section_name: None,
+                        port: None,
+                    },
+                    controller_name: STATUS_CONTROLLER_NAME.to_string(),
+                    conditions: vec![condition],
+                }
+            })
+            .collect();
+        let status = gateway::HttpRouteStatus {
+            inner: gateway::RouteStatus {
+                parents: parent_statuses,
+            },
+        };
+        let patch = serde_json::json!({
+            "apiVersion": POLICY_API_VERSION,
+            "kind": "HTTPRoute",
+            "name": route_id.name,
+            "status": status,
+        });
+
+        // Patch the HTTPRoute with its status.
+        let api = k8s::Api::<k8s::policy::HttpRoute>::namespaced(
+            self.client.clone(),
+            &route_id.namespace,
+        );
+        if let Err(error) = api
+            .patch_status(&route_id.name, &patch_params, &k8s::Patch::Merge(patch))
+            .await
+        {
+            tracing::error!(namespace = %route_id.namespace, name = %route_id.name, %error, "Failed to patch HTTPRoute");
+        }
+    }
+
+    async fn process_server_update(&mut self, patch_params: k8s::PatchParams) {
+        let route_ids: Vec<ResourceId> = self.index.read().http_routes.keys().cloned().collect();
+        for route_id in route_ids {
+            self.process_http_route_update(route_id, patch_params.clone())
+                .await;
+        }
+    }
+}
+
+impl Index {
+    pub fn shared(updates: UnboundedSender<Update>) -> SharedIndex {
+        Arc::new(RwLock::new(Self {
+            updates,
+            http_routes: HashMap::new(),
+            servers: HashSet::new(),
+        }))
+    }
+
+    fn update_http_route(&mut self, route_id: ResourceId, route_binding: RouteBinding) -> bool {
+        match self.http_routes.entry(route_id) {
+            Entry::Vacant(entry) => {
+                entry.insert(route_binding);
+            }
+            Entry::Occupied(mut entry) => {
+                if *entry.get() == route_binding {
+                    return false;
+                }
+                entry.insert(route_binding);
+            }
+        }
+        true
+    }
+}
+
+impl kubert::index::IndexNamespacedResource<k8s::policy::HttpRoute> for Index {
+    fn apply(&mut self, resource: k8s::policy::HttpRoute) {
+        let namespace = resource
+            .namespace()
+            .expect("HTTPRoute must have a namespace");
+        let name = resource.name_unchecked();
+        let id = ResourceId::new(namespace.clone(), name.clone());
+
+        let route_binding = match RouteBinding::try_from(resource) {
+            Ok(binding) => binding,
+            Err(error) => {
+                tracing::info!(%namespace, %name, %error, "Ignoring HTTPRoute");
```

Comment:
Will this be logged often? Might be worth doing it at a debug level?

Reply:
Nope this shouldn't be logged often. We should usually be able to convert an HTTPRoute to a `RouteBinding`. It's `info` here because we do the same in the policy controller's index. I suppose it could be `error` as well, but again just keeping it the same as the other index code.